server.js

```diff
@@ -6892,6 +6892,126 @@ cache((data, match, sendBadge, request) => {
   });
 }));
 
+// nsp for npm packages
+camp.route(/^\/nsp\/npm\/(@([^/]+)\/)?[^/]+(\/[^/]+)?\.(svg|png|gif|jpg|json)$/, cache((data, match, sendBadge, request) => {
+  // A: /nsp/npm/:package.:format
+  // B: /nsp/npm/:package/:version.:format
+  // C: /nsp/npm/@:scope/:package.:format
+  // D: /nsp/npm/@:scope/:package/:version.:format
+  const pieces = match.input.split('/');
+  const badgeData = getBadgeData('nsp', data);
+  const nspRequestOptions = {
```

Comment:
It's difficult to understand how data is flowing between the main body of this badge provider and the local functions. To make this clearer, could you declare this `nspRequestOptions ` structure inside `getNspResults` instead? You could pass in the package name as a parameter, like you're doing with the package version.